From cdc31e82c3c7c51fe520525f7d8b167df11975be Mon Sep 17 00:00:00 2001
From: Richard Muzik <richard.muzik@turris.com>
Date: Tue, 14 Oct 2025 10:55:11 +0200
Subject: [PATCH] syslog-ng: fix compilation for aarch64

This is more of a hack to fix compilation for aarch64 target.

Fixes:
```
In file included from iv_private.h:35,
                 from iv_event.c:27:
iv_private_posix.h:110:49: error: field 'ring' has incomplete type
  110 |                         struct io_uring         ring;
      |                                                 ^~~~
iv_private_posix.h:113:57: error: field 'ts' has incomplete type
  113 |                         struct __kernel_timespec        ts;
      |                                                         ^~
```

Note that this happens if liburing is compiled before the syslog-ng.
---
 admin/syslog-ng/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/admin/syslog-ng/Makefile b/admin/syslog-ng/Makefile
index 885954d90..cb357d550 100644
--- a/admin/syslog-ng/Makefile
+++ b/admin/syslog-ng/Makefile
@@ -109,7 +109,8 @@ CONFIGURE_ARGS +=  \
 
 CONFIGURE_VARS += \
 	$(if $(CONFIG_HOST_OS_MACOS),PATH=$(FAKEUNAME_PATH):$(PATH)) \
-	LIBDBI_CFLAGS="-I$(STAGING_DIR)/usr/include"
+	LIBDBI_CFLAGS="-I$(STAGING_DIR)/usr/include" \
+	ac_cv_search_io_uring_queue_init=no
 
 define Package/syslog-ng/install
 	cd $(PKG_BUILD_DIR); make DESTDIR=$(1) install
-- 
2.51.0

